### DISCUSSION TITLE
EMFile: Too many open files

### DISCUSSION THREAD

**Discussion by divillysausages** [serverless/serverless#10160] [Q&A] [answered]
I've been running into this nearly every time I run `deploy` locally (Windows 10).

Looking at #3249 and #9073 it seems that it's a long-term issue but one that had been addressed.

My serverless version:

```
Framework Core: 2.59.0
Plugin: 5.4.4
SDK: 4.3.0
Components: 3.17.0
```

I'm using the v3 version of the AWS SDK, which, while breaking different packages into separate modules, seems to be more prone to hitting it. My dependencies aren't (in my opinion) excessive (devDependencies should be stripped by default, correct?):

```
"dependencies": {
    "@aws-sdk/client-dynamodb": "^3.31.0",
    "@aws-sdk/client-s3": "^3.31.0",
    "@aws-sdk/client-sns": "^3.31.0",
    "@aws-sdk/client-sqs": "^3.31.0",
    "@aws-sdk/util-dynamodb": "^3.31.0",
    "jsonwebtoken": "^8.5.1",
    "node-fetch": "^2.6.2",
    "seedrandom": "^3.0.5",
    "ulid": "^2.3.0"
  }
```

As I'm on Windows, the usual solutions of changing `ulimit` don't work, and `graceful-fs` is a bit hit and miss (where am I supposed to plug it in to global patch the `fs` module?). If I run `deploy` right after starting my laptop, it seems to have a better chance of passing, but with a 10-15 minute wait time before knowing if it's successful, it's more than irritating.

In short, does anyone have a definitive solution for this on Windows, or can anyone familiar with the project comment on whether the fixes for the two bug reports listed above are still valid?

I'm posting it here as a general question so I can gauge if it requires an actual bug report - considering that similar reports have been closed multiple since over nearly 4 years.

Thanks

**Comment by divillysausages** [comment]
I really need to know how people are approaching this issue. Even with `graceful-fs`, I'm getting a `Maximum call stack size exceeded` error.

How do I deploy on Windows? Is it possible to package each service (`package.individually` is set to `true`) one after the other instead of all at once?

**Comment by divillysausages** [comment, marked as answer]
It seems that with a combination of `graceful-fs` and upgrading to the latest sdk (2.65.0) seems to work.

Based on this comment: https://github.com/serverless/serverless/issues/10161#issuecomment-954827939, they seem to have handled the edge cases and 2.65.0 is the latest version that supported that.

Be careful though, as `serverless` seems to use a local copy of serverless that comes in through `node_modules` somewhere rather than the globally installed package.
